packages/hardhat-chai-matchers/test/bigNumber.ts

```diff
@@ -877,20 +877,32 @@ describe("BigNumber matchers", function () {
 
           // failed assertions
           expect(() => expect([a]).to.not.deep.equal([b])).to.throw(
-            AssertionError
+            AssertionError,
+            // [\s\S]* is used here instead of .* so that newlines are included
+            /expected \[[\s\S]*\] to not deeply equal \[[\s\S]*\]/
```

Comment:
Is it not possible (or practical) to actually put the value in here? eg `${a}`? `/expected \[.*${b}.*\] to not deeply equal \[.*${a}.*\]/s`?

Reply:
This is a bit hard, because the value in the error message is sometimes `1`, sometimes `1n`, and sometimes `BigNumber { value: "1" }` (or something like that). So I opted for the lazy approach of checking that the error looks _somewhat_ correct.